drizzlepac/devutils/comparison_tools/compare_sourcelists.py

```diff
@@ -300,135 +399,166 @@ def computeLinearStats(matchedRA,plotGen,diffMode,plot_title,plotfile_prefix,ver
 
-        ax1.axvline(x=np.mean(diffRA)+np.std(diffRA), color='g', linestyle=':')
-        ax1.axvline(x=np.mean(diffRA)-np.std(diffRA), color='g', linestyle=':')
         ax1.set_xlabel("$\Delta %s$" % (xAxisString))
         ax1.set_ylabel("Number of matched sources")
 
         ax2 = ax1.twinx()
         ax2.hist(diffRA, bins=bins, cumulative=-1, density=True, histtype='step', color='r')
-        ax2.set_ylabel("Fraction of all matched sources",color='r')
+        ax2.set_ylabel("Fraction of all matched sources", color='r')
         for tl in ax2.get_yticklabels():
             tl.set_color('r')
 
+        log.info("\n")
+        for log_line in log_output_string_list:
+            if log_line == "\n":
+                log.info("")
+            else:
+                log.info(log_line)
+
         if plotGen == "screen":
             plt.show()
+            plotFileName = ""
+            stat_file_name = ""
         if plotGen == "file":
             # Put timestamp and plotfile_prefix text string in lower left corner below plot
             timestamp = "Generated {}".format(datetime.now().strftime("%m/%d/%Y %H:%M:%S"))
             plt.text(0.0, -0.081, timestamp, horizontalalignment='left', verticalalignment='center', fontsize=5,
                      transform=ax1.transAxes)
             plt.text(0.0, -0.105, plotfile_prefix, horizontalalignment='left', verticalalignment='center', fontsize=5,
                      transform=ax1.transAxes)
-            plotFileName = "{}_{}.pdf".format(plotfile_prefix,plot_title.replace(" ","_"))
+            plotFileName = "{}_{}.pdf".format(plotfile_prefix, plot_title.replace(" ", "_"))
             if plotFileName.startswith("_"):
-                plotFileName=plotFileName[1:]
+                plotFileName = plotFileName[1:]
             plt.savefig(plotFileName)
             plt.close()
-            log.info("{} plot saved to file {}.".format(fullPlotTitle, plotFileName))
-    log.info("\n")
-    return(regTestStatus+out_stats)
+            # generate second pdf page with statistics
+            stat_file_name = plotFileName.replace(".pdf", "_stats.pdf")
+            fig = plt.figure(figsize=(11, 8.5))
+            fig.text(0.5, 0.87, fullPlotTitle[:-1] + " statistics", transform=fig.transFigure, size=12, ha="center")
+            stat_text_blob = ""
+            for log_line in log_output_string_list:
+                if log_line != "\n":
+                    stat_text_blob += log_line + "\n"
+                else:
+                    stat_text_blob += "\n"
+            stat_text_blob += "\n" + timestamp + "\n"
+            stat_text_blob += plotfile_prefix
+            fig.text(0.5, 0.5, stat_text_blob, transform=fig.transFigure, size=10, ha="center", va="center",
+                     multialignment="left", family="monospace")
+            fig.savefig(plotFileName.replace(".pdf", "_stats.pdf"))
+            plt.close()
```

Comment:
This section of code could be turned into a simple generic function with 'log_output_string_list' and 'plotFileName' as the inputs. The new function would not only be called here, but also from 'computeFlagStats' (line 339-354?) as well and limit the amount of duplicate code in this module.